### DISCUSSION TITLE
use congestion control goodness from ion into pion

### DISCUSSION THREAD

**Discussion by uharband** [pion/webrtc#1523] [Q&A]
Hi!

thanks for the great pion project! 
i'm experimenting using pion webrtc for sending pre-encoded content like from a local file to a browser. i'm interested in congestion control since i'm experiencing a high rate of packet loss.
now, i know its 'on the way' and planned for the next version but i see the ion project already has a lot implemented in the congestion control area under the [rtc ](https://github.com/pion/ion/tree/master/pkg/rtc) package. 
my question is how can i 'carve out' this and plug it into my pion based webrtc project. 
i'm currently using the pion api for adding tracks and then i typically call the [WriteSample ] (https://github.com/pion/webrtc/blob/master/track.go#L182) API on the track object. i can't figure out quite exactly if and how i can adapt the current pion/webrtc APIs to the ion/rtc APIs to get all the congestion control features implementation
i'm only using one direction, from pion to browser

Thanks so much!

Uzi

**Comment by Sean-Der** [comment]
Hey @uharband 

We are trying to do that today! In `pion/webrtc@v3` we are adding the concept of `Interceptors` you can read about it [here](https://github.com/pion/webrtc-v3-design/issues/34) and [here](https://github.com/pion/webrtc-v3-design/issues/36) the idea is that we are going to have everything be pluggable.

If you are interested in this I would love your input! We have a PR [here]( https://github.com/pion/webrtc/pull/1504) and would love your help testing and getting all the goodness in. Once it is in users could distribute their own interceptors, we could have competing congestion controllers that live completely outside of Pion (I think that is super cool)
